lib/inputstreamhelper/__init__.py

```diff
@@ -546,6 +546,61 @@ def _chromeos_config(self):
 
         return devices
 
+    def _backup_libwv(self):
+        """Creates backup of installed lib, keeping multiple backups according to setting 'backups'"""
+        backups = int(get_setting('backups', '4'))
+        backup_dir = os.path.join(self._ia_cdm_path(), 'backup')
```

Comment:
Not sure it's a good idea to make backups to the InputStream Adaptive CDM directory because InputStream Adaptive add-on uses this directory in a way we can't control.
I think it's better to save all backups in our own userdata directory and only copy a single CDM when needed.

Reply:
@mediaminister It is part of the library search path, but based on name AFAIK. So that shouldn't be a problem if we stick to a proper naming plan (not something that could accidentally be picked up). In fact on Linux it is common to have multiple versions next to each other (and a symlink for an unversioned name). Although I wouldn't use the Unix convention to avoid it being picked up accidentally ;-)

Reply:
That said, I wouldn't mind if we decide to move it elsewhere. I just think it is nicer if it stays together. (and the hardlink shows which one is being used)

Reply:
> Not sure it's a good idea to make backups to the InputStream Adaptive CDM directory because InputStream Adaptive add-on uses this directory in a way we can't control.
> I think it's better to save all backups in our own userdata directory and only copy a single CDM when needed.

So keeping the license file and widevine_config.json in the cdm dir is fine, but a backup dir isn't? You're the one who put these files there in #95 @mediaminister .

BTW I decided to put the backups in an extra dir to avoid confusing potential fuzzy searches or similar, I think it should be rather safe there.

Reply:
Mozilla Firefox also stores the `LICENSE.txt` and `manifest.json` in the same directory as `libwidevinecdm.so`...

I'm not sure if it's fine or even useful that `manifest.json` is renamed to `widevine_config.json`, but maybe @emilsvennesson remembers why this is done: https://github.com/emilsvennesson/script.module.inputstreamhelper/commit/2641cadfcb3aad741138f827f0b59bfc4bf2e2a0

There is also a [Kodi add-on rule](https://kodi.wiki/view/Add-on_rules) that states the following:
> add-ons should store all their data in their own subfolder inside the addon_data directory. Access (read/write/delete) to any other files / folders is not allowed by default.

If a Kodi user decides to remove the InputStreamHelper add-on, the backup dir will exist forever...

Those are the reasons why I have my doubts.

Reply:
Alright, that clears it up. I'll move the backup dir.